2024/incoming/a-time-tested-powerhouse-for-processing-xml.pod

```diff
@@ -0,0 +1,168 @@
+
+=begin code
+use strict;
+use warnings;
+use diagnostics;
+use XML::Twig;
+
+sub check_guest {
+  my $id = shift;
+  my $password = shift; 
+
+  my $twig = XML::Twig -> new();
+  $twig -> parsefile( "/hotelpms/guests.xml") || # handle error;
+
+}
+=end code
+
+In order to access guest tags, we first need to obtain guestList, which is the root of this
+XML document. The C<<$twig->root>> method returns the direct parent of all other elements. On the
+other hand, the C<children> method returns list of elements. The method can take an optional 
+argument. If a string is passed to the method, XML elements that match the string will be fetched;
+otherwise, all elements of the current root will be fetched in document order. The returned list 
+contains elements that are instances of the C<XML::Twig::Elt> class. So, this means you can perform 
+any operations that the C<XML::Twig::Elt> class allows. 
+
+As you can see, information we are looking for is the attributes of the XML element tag. To reach 
+the name of a person we must access name attribute. C<XML::Twig::Elt> has various functions on
+attributes. The C<att> method will do the job. To note that, because the XML we have is automatically
+generated by another system, so we haven't checked if the element has the attribute. To check if an
+element has a specific attribute, you can call the C<att_exist> it will return true if the attribute
+exists for the element, false otherwise.
+
+=begin code
+sub check_guest {
+  # previous lines above
+
+  my @guests = $twig->root->children('guest');
+  # $guests[0]->att('name') will return Michael
+}
+=end code
+
+If we put all the information we’ve discussed together, the function would look like this.
+
+=begin code
+
+sub check_guest {
+  my $id = shift;
+  my $password = shift; 
+
+  my $twig = XML::Twig -> new();
+  $twig -> parsefile( "/hotelpms/guests.xml") || return 0; # handle error
+  my @guests = $twig->root->children('guest');
+  
+  foreach my $guest (@guests) {
+    return 1 if (($guest -> att('id') eq $id) && ($guest -> att('wifiKey') eq $password));
+  }
+
+  return 0;
+}
+
+=end code
+
+=head2 Parsing XML Response
+Suppose that there are servers somewhere in this universe. We don't know exactly when the answer
+will be returned it changes depending on the data of our request, but we know the exact XML response
+format. It is said that this server provides answers about the Earth's past based on what it 
+observes. So, demanding information related to an much earlier date will take much longer to travel
+due to the speed of light and it is open anyone to send request. A researcher wants to find out 
+why the stones at Gobeklitepe were buried. Researcher knows that when the answer reach researcher
+will not be alive. All conditions have been maintained for the message to return, and a system was 
+designed to await all requests by some world organization. The language selected was Perl.
+
+Here is excepting response format:
```

Comment:
Not quite sure what you mean by `excepting`?

Reply:
I should have written `expected` 😅 I'am going to correct it